File: src/main/java/com/back/together02be/chart/dto/response/CandleRes.java
package com.back.together02be.chart.dto.response;

import java.math.BigDecimal;

import com.back.together02be.chart.dto.Candle;

public record CandleRes(
	String time,
	BigDecimal open,
	BigDecimal high,
	BigDecimal low,
	BigDecimal close,
	Long volume
) {
	public static CandleRes from(Candle candle) {
		return new CandleRes(
			candle.date().toString(), // yyyy-MM-dd
			candle.open(), candle.high(), candle.low(), candle.close(), candle.volume()
		);
	}
}

File: src/main/kotlin/com/back/together02be/chart/dto/response/CandleRes.kt
package com.back.together02be.chart.dto.response

import com.back.together02be.chart.dto.Candle
import java.math.BigDecimal

data class CandleRes(
    val time: String,
    val open: BigDecimal,
    val high: BigDecimal,
    val low: BigDecimal,
    val close: BigDecimal,
    val volume: Long
) {
    companion object {
        fun from(candle: Candle): CandleRes =
            CandleRes(
                time = candle.date.toString(),  // yyyy-MM-dd
                open = candle.open,
                high = candle.high,
                low = candle.low,
                close = candle.close,
                volume = candle.volume
            )
    }
}
